src/state/events.cpp

```diff
@@ -0,0 +1,43 @@
+#include "app.hpp"
+
+#define G_LOG_DOMAIN "genie::state::events"
+
+namespace genie {
+namespace state {
+namespace events {
+
+// ### Conversation Events ###
+
+InputFrame::InputFrame(AudioFrame *frame) : frame(frame) {}
+InputFrame::~InputFrame() {
+  // NOTE We do **NOT** free the frame on destruction -- it's handed off to the
+  //      STT instance, where it may be queued for later transmission.
```

Comment:
This is dangerous memory management, eg if the input frame event is not handled by the STT. It would be better to have a "steal_frame()" method that moves the AudioFrame out / sets the pointer to null.

Reply:
Yup, it is dangerous. Need to learn how that move stuff works then I'll switch it over to that

Reply:
Addressed in cc86f8646790a149bde3512c8b58e2c61fce9123